006-strimzi-ui.md

```diff
@@ -114,6 +114,28 @@ Finally, one capability which may be of interest (which may have bearing on how
 
 In all of these cases, capabilities can be added in a prioritised order, and should be added in a phased manner themselves (for example, add a view of the brokers and their configuration, then the ability to modify select configuration).
 
+#### Proposed deployment
+
+This UI could be deployed as a part of Strimzi as follows:
+
+![Suggested deployment](./images/006-deployment.png)
+
+Where:
+
+- The UI is deployed standalone, akin to how the Kafka bridge, Kafka Connect are currently for example (so a new CRD would be required, with configuration needed to reference which cluster(s) to interact with).
+- The UI's spec would contain 0 to N Kafka clusters, which: 
+  - Contain 1 or more 'backend' entries. These represent backend sources of data which can be surfaced in the UI.
+  - One of these entries will be the `strimzi-api` (ie each Kafka kind will contain its own api server).
```

Comment:
how the backend is deployed? Adding some fields in the `Kafka` resource or always starting from the backend entries in the new Web UI resource? I am thinking about the user experience in configuring the UI where they would need to create a new resource for the actual Web UI (on which I could agree) but even adding some more information to the `Kafka` resource as far as I understood in your proposal.

Reply:
I am thinking the `kafka` resource would deploy (if configured - ie it could be optional, but would defer to the ongoing api proposal/discussions) a deployment which is the strimzi-api. It could be as a part of this deployment (say as a piece of status information) output the configuration needed for the UI proposed here. A user could then take that, and insert it at UI deploy time to their CR. I do see the backend needing to exist/be deployed before the UI could be deployed myself - but this could be achieved in a number of different ways. Would you think a more automated approach would be preferable?